examples/Transfer_Learning/basic_tl.py

```diff
@@ -0,0 +1,177 @@
+# [NumericalContinuousParameters](baybe.parameters.numerical.NumericalContinuousParameter),
+# we use discrete parameters here because it lets us interpret the percentages shown in
+# the final plot directly as the proportion of candidates for which there were target
+# values revealed by the training function.
+# ```
+
+# Next, we define a
+# [TaskParameter](baybe.parameters.discrete.TaskParameter) to encode the task context,
+# which allows the model to establish a relationship between the training data and
+# the data collected during the optimization process.
+# Because we want to obtain recommendations only for the test function, we explicitly
+# pass the `active_values` keyword.
+
+task_param = TaskParameter(
+    name="Function",
+    values=["Test_Function", "Training_Function"],
+    active_values=["Test_Function"],
+)
+
+# With the parameters at hand, we can now create our search space.
+
+parameters = [*discrete_params, task_param]
+searchspace = SearchSpace.from_product(parameters=parameters)
+
+# (Lookup)=
+### Generating Lookup Tables
+
+# We generate two lookup tables containing the target values of both test
+# functions at the given parameter grid.
+# Parts of one lookup serve as the training data for the model.
+# The other lookup is used as the loop-closing element, providing the target values of
+# the test functions on demand.
+
+grid = np.meshgrid(*[p.values for p in discrete_params])
+
+lookups: Dict[str, pd.DataFrame] = {}
+for function_name, function in test_functions.items():
+    lookup = pd.DataFrame({f"x{d}": grid_d.ravel() for d, grid_d in enumerate(grid)})
+    lookup["Target"] = lookup.apply(function, axis=1)
+    lookup["Function"] = function_name
+    lookups[function_name] = lookup
+lookup_training_task = lookups["Training_Function"]
+lookup_test_task = lookups["Test_Function"]
+
+### Simulation Loop
+
+# We now simulate campaigns for different amounts of training data unveiled,
+# to show the impact of transfer learning on the optimization performance:
+
+results: List[pd.DataFrame] = []
+for p in (0.01, 0.02, 0.05, 0.08, 0.2):
+    campaign = Campaign(searchspace=searchspace, objective=objective)
+    initial_data = [lookup_training_task.sample(frac=p) for _ in range(N_MC_ITERATIONS)]
+    result_fraction = simulate_scenarios(
+        {f"{100*p}": campaign},
+        lookup_test_task,
+        initial_data=initial_data,
+        batch_size=BATCH_SIZE,
+        n_doe_iterations=N_DOE_ITERATIONS,
+    )
+    results.append(result_fraction)
+
+# For comparison, we also optimize the function without using any initial data:
+
+result_fraction = simulate_scenarios(
+    {"0.0": Campaign(searchspace=searchspace, objective=objective)},
+    lookup_test_task,
+    batch_size=BATCH_SIZE,
+    n_doe_iterations=N_DOE_ITERATIONS,
+    n_mc_iterations=N_MC_ITERATIONS,
+)
+results = pd.concat([result_fraction, *results])
+
+# All that remains is to visualize the results:
+
+themes = json.load(open("plotting_themes.json"))
```

Comment:
One last comment: I think you'll change this part anyway once the other PR is merged, but two things to keep in mind:
* I think the plotting_themes.json should NOT stay at the repo root folder. This place is for repo configuration stuff only (i.e. pyproject.toml etc) but not actual content.
* Whereever you place it, make sure that the examples run irrespective from where you call them. At the moment, it would only work when python is invoked from the repository root

Reply:
Regarding the first point: Where should we then put it? I do not really see a good place other then maybe the static folder.

Regarding the second point: True, but what is the best way to ensure that the file is found? How can we even do that, given that the current working directory from which python is being run can be arbitrary? Only solution I see is via having yet another parameter to the plotting function.

Reply:
This was fixed in the prevous PR